### PR TITLE
Improve now playing banner

### PR DESCRIPTION
**Pull request by nukooo**
# Rewrite banners as independent objects
The current implementation is fragile. Because updates are performed sequentially, an inaccessible API blocks all radios from updating until the fetch times out. Likewise, an unhandled exception in an unmarshalfn will abort the update for all stations.

The new implementation: 
- Makes updating banners concurrent and independent
- Allows easier addition, removal, and rearrangement of banners by eliminating the implicit coupling between declaration order and bit field index in the nowPlaying option
- Improves type safety

# Horizontal banner option
Stacked banners waste screen space and many users prefer the original layout. While this can be fixed by the user with custom CSS, an official option is more convenient.

This option is enabled by default to match how the banner used to be displayed. As pointed out in [PR #1241][1], horizontal banners can be problematic on small screens. I believe the majority of users have enough screen width that this isn't a problem. Those that don't can easily turn it off.

# Eden default streamer
Eden Radio lacks a bot streamer name. This is inconsistent with the other banners so I've added a default.

[1]: https://github.com/bakape/meguca/pull/1241#issuecomment-703129070

### CODE REVIEW

File: client/main.ts [hunk 1]
```diff
@@ -28,24 +28,29 @@ async function start() {
...
+			break
+		case "eden":
+			localStorage.setItem("eden", "true")
+			break
+		case "both":
+			localStorage.setItem("radio", "true")
+			localStorage.setItem("eden", "true")
+			break
+		default:
+			const flags = parseInt(nowPlaying, 10)
```
Comment (bakape) [inline client/main.ts:46]: It's amazing how this still works with `nowPlaying=null`.

**Review by bakape — APPROVED** [review]

**Comment by bakape** [comment]
Please resolve conflicts and I'll merge it.

File: client/main.ts (hunk 1 above)
Comment (nukooo) [reply client/main.ts]: NaN is converted to 0 in bitwise contexts. That migration should only run if nowPlaying exists however. Good catch.

**Comment by bakape** [comment]
Thanks!

**Pull request merged**